File: ame-core/src/main/kotlin/com/agenticmobile/ame/AmeConformanceCli.kt
package com.agenticmobile.ame

import kotlin.system.exitProcess

fun main(args: Array<String>) {
    if (args.isEmpty()) {
        System.err.println("Usage: ame-conformance <file.ame>")
        exitProcess(1)
    }
    val input = java.io.File(args[0]).readText()
    val parser = AmeParser()
    val tree = parser.parse(input) ?: run {
        System.err.println("Parse returned null")
        exitProcess(2)
    }
    println(AmeSerializer.toJson(tree))
}

File: ame-flutter/bin/ame_conformance_cli.dart
import 'dart:io';

import 'package:ame_flutter/ame_flutter.dart';

void main(List<String> args) {
  if (args.isEmpty) {
    stderr.writeln('Usage: dart run ame_flutter:ame_conformance_cli <file.ame>');
    exit(1);
  }

  final input = File(args[0]).readAsStringSync();
  final parser = AmeParser();
  final tree = parser.parse(input);

  if (tree == null) {
    stderr.writeln('Parse returned null');
    exit(2);
  }

  print(AmeSerializer.toJson(tree));
}
